common/mem.test.c

```diff
@@ -159,6 +159,38 @@ test_strdup_strndup(UNUSED const MunitParameter params[], UNUSED void *data)
 	return MUNIT_OK;
 }
 
+static MunitResult
+test_mem_memcpy(UNUSED const MunitParameter params[], UNUSED void *data)
+{
+	unsigned char a[257];
+	munit_rand_memory(sizeof(a), a);
+
+	// memcpy duplicates memory
+	unsigned char *b = mem_memcpy(a, sizeof(a));
+	munit_assert_not_null(b);
+	munit_assert_memory_equal(sizeof(a), a, b);
+
+	// b is in a different memory chunk than a and can be modified
+	b[8] = ~b[8];
+	munit_assert_memory_not_equal(sizeof(a), a, b);
+	mem_free(b);
+
+	// memcpy copies entire memory
+	unsigned char *c = mem_memcpy(a, sizeof(a));
+	munit_assert_not_null(c);
+	munit_assert_memory_equal(sizeof(a), a, c);
+	mem_free(c);
+
+	// memcpy cuts
+	c = mem_memcpy(a, 129);
+	munit_assert_not_null(c);
+	munit_assert_memory_not_equal(129, a, c);
```

Comment:
This line fails in the Pipeline, please fix

Reply:
done